2011-05-22  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/unix/sysv/linux/bits/resource.h (RLIMIT_RTTIME): Add.
	* sysdeps/unix/sysv/linux/sparc/bits/resource.h: Likewise.
	* ports/sysdeps/unix/sysv/linux/mips/bits/resource.h: Likewise.
	* ports/sysdeps/unix/sysv/linux/alpha/bits/resource.h: Likewise.

diff a/sysdeps/unix/sysv/linux/bits/resource.h b/sysdeps/unix/sysv/linux/bits/resource.h
---
 ports/sysdeps/unix/sysv/linux/alpha/bits/resource.h |    8 +++++++-
 ports/sysdeps/unix/sysv/linux/mips/bits/resource.h  |    8 +++++++-
 sysdeps/unix/sysv/linux/bits/resource.h             |    8 +++++++-
 sysdeps/unix/sysv/linux/sparc/bits/resource.h       |    8 +++++++-
 4 files changed, 28 insertions(+), 4 deletions(-)

--- a/sysdeps/unix/sysv/linux/bits/resource.h
+++ b/sysdeps/unix/sysv/linux/bits/resource.h
@@ -100,7 +100,13 @@
   __RLIMIT_RTPRIO = 14,
 #define RLIMIT_RTPRIO __RLIMIT_RTPRIO
 
-  __RLIMIT_NLIMITS = 15,
+  /* Maximum CPU time that a process scheduled under a real-time
+     scheduling policy may consume without making a blocking
+     system call.  */
+  __RLIMIT_RTTIME = 15,
+#define RLIMIT_RTTIME __RLIMIT_RTTIME
+
+  __RLIMIT_NLIMITS = 16,
   __RLIM_NLIMITS = __RLIMIT_NLIMITS
 #define RLIMIT_NLIMITS __RLIMIT_NLIMITS
 #define RLIM_NLIMITS __RLIM_NLIMITS
--- a/sysdeps/unix/sysv/linux/sparc/bits/resource.h
+++ b/sysdeps/unix/sysv/linux/sparc/bits/resource.h
@@ -100,7 +100,13 @@
   __RLIMIT_RTPRIO = 14,
 #define RLIMIT_RTPRIO __RLIMIT_RTPRIO
 
-  __RLIMIT_NLIMITS = 15,
+  /* Maximum CPU time that a process scheduled under a real-time
+     scheduling policy may consume without making a blocking
+     system call.  */
+  __RLIMIT_RTTIME = 15,
+#define RLIMIT_RTTIME __RLIMIT_RTTIME
+
+  __RLIMIT_NLIMITS = 16,
   __RLIM_NLIMITS = __RLIMIT_NLIMITS
 #define RLIMIT_NLIMITS __RLIMIT_NLIMITS
 #define RLIM_NLIMITS __RLIM_NLIMITS
--- a/ports/sysdeps/unix/sysv/linux/mips/bits/resource.h
+++ b/ports/sysdeps/unix/sysv/linux/mips/bits/resource.h
@@ -100,7 +100,13 @@
   __RLIMIT_RTPRIO = 14,
 #define RLIMIT_RTPRIO __RLIMIT_RTPRIO
 
-  __RLIMIT_NLIMITS = 15,
+  /* Maximum CPU time that a process scheduled under a real-time
+     scheduling policy may consume without making a blocking
+     system call.  */
+  __RLIMIT_RTTIME = 15,
+#define RLIMIT_RTTIME __RLIMIT_RTTIME
+
+  __RLIMIT_NLIMITS = 16,
   __RLIM_NLIMITS = __RLIMIT_NLIMITS
 #define RLIMIT_NLIMITS __RLIMIT_NLIMITS
 #define RLIM_NLIMITS __RLIM_NLIMITS
--- a/ports/sysdeps/unix/sysv/linux/alpha/bits/resource.h
+++ b/ports/sysdeps/unix/sysv/linux/alpha/bits/resource.h
@@ -100,7 +100,13 @@
   __RLIMIT_RTPRIO = 14,
 #define RLIMIT_RTPRIO __RLIMIT_RTPRIO
 
-  __RLIMIT_NLIMITS = 15,
+  /* Maximum CPU time that a process scheduled under a real-time
+     scheduling policy may consume without making a blocking
+     system call.  */
+  __RLIMIT_RTTIME = 15,
+#define RLIMIT_RTTIME __RLIMIT_RTTIME
+
+  __RLIMIT_NLIMITS = 16,
   __RLIM_NLIMITS = __RLIMIT_NLIMITS
 #define RLIMIT_NLIMITS __RLIMIT_NLIMITS
 #define RLIM_NLIMITS __RLIM_NLIMITS
